redbot/cogs/downloader/repo_manager.py

```diff
@@ -629,6 +630,16 @@ async def clone(self) -> Tuple[Installable, ...]:
             git_command = ProcessFormatter().format(
                 self.GIT_CLONE_NO_BRANCH, url=self.url, folder=self.folder_path
             )
+
+        async with aiohttp.ClientSession() as session:
+            async with session.head(self.url, allow_redirects=True) as response:
+                if response.status == 404:
+                    raise errors.MissingRemoteGitRepo("The repository was not found.")
+                elif 400 <= response.status < 500 or 500 <= response.status < 600:
+                    raise errors.MissingRemoteGitRepo(
+                        f"The reposity returned an error code: {response.status}."
```

Comment:
Typo in "repository"
```suggestion
                        f"The repository returned an error code: {response.status}."
```